study-group/Toast-Implementation-Studies.md

```diff
@@ -70,25 +70,164 @@ The `<paper-toast>` holds onto information about how it was dismissed the last t
+        <ToastContainer />
+      </div>
+    );
+  }
+}
+```
+
+`react-toastify` uses the explicit component pattern, by having the developer render a React `ToastContainer` component (the recommendation is to put this in the application root).
+This single component will render all toasts called from JavaScript, and can be configured with global options.
+However, the library allows the developer a lot of freedom, supporting multiple `ToastContainer`s as well as no `ToastContainer`, though one is recommended.
+When the `toast()` is called with a message and a props object, it is displayed in the toast container, with the toast props overriding the `ToastContainer` props, like so:
+
+```js
+class App extends Component {
+  tenSecondToast = () => toast("This toast will self destruct in 10...", { autoClose: 10000 });
+  normalToast = () => toast("This is a normal toast");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.tenSecondToast}>Close after 10 seconds</button>
+        <button onClick={this.normalToast}>Close after default amount of time</button>
+        <ToastContainer autoClose={5000} />
+      </div>
+    );
+  }
+}
+```
+
+The first button creates a toast that dismisses in 10 seconds, overriding the `ToastContainer` default.
+The second provides no `autoClose` value, so it dismisses in the default 5 seconds.
+For some of the more common controls, the toast can be configured like so:
+
+```js
+class App extends Component {
+  topRightWarning = () => toast('Warning!', {
+    type: toast.TYPE.WARNING,
+    position: toast.POSITION.TOP_RIGHT,
+    onClose: ({ props }) => window.alert('I warned you') // props: all props passed to rendered toast component
+  });
+
+  render() {
+    return (
+      <button onClick={this.topRightWarning}>Warn me</button>
+      <ToastContainer autoClose={5000} />
+    );
+  }
+}
+```
+
+`react-toastify` supports built-in themes for toasts, such as `success`, `info`, and `warning`, among others.
+It also provides intuitive positioning, that can be set from the toast or configured on the `ToastContainer`.
+Additionally, callbacks like `onClose` and `onOpen` allow the toast to have intelligent effects when the user interacts with it.
+
+### Notable Features & Details
+
+#### Toast ID
+the `toast()` function returns an ID of the toast it creates, which can be used to act upon that toast.
+For instance, with the toast ID the toast can be dismissed, updated, and checked to see if it's active.
+
+#### Mobile
+On mobile toasts will automatically take all available width, and all toasts are swipeable / draggable to dismiss.
+
+#### Styling
+Library provides fields `className`, `toastClassName`, and, `progressClassName` for the developer to pass in CSS classes.
+
+#### Render Custom Component
+To customize the view the `toast` function can take in a React component, which will receive the function required to close the toast as props (though it won't get the toast ID).
```

Comment:
Can you rephrase this? I don't know what this means.

Reply:
This was unclear, rephrasing now